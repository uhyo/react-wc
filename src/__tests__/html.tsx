import { render, screen } from "@testing-library/react";
import { html } from "..";

describe("html", () => {
  it("no slot", () => {
    const Hello = html`
      <style>
        div {
          font-size: 100px;
        }
      </style>
      <div>Hello</div>
    `;

    render(<Hello />);

    screen.
  });
});
