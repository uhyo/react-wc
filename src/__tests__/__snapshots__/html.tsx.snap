// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html Named slot values array 1`] = `"<b slot=\\"child\\">Hello</b><b slot=\\"child\\">val is foobar</b><b slot=\\"child\\">val is wow</b>"`;

exports[`html Named slot values boolean 1`] = `"<wc-text slot=\\"child\\"></wc-text>"`;

exports[`html Named slot values class component 1`] = `"<b slot=\\"child\\">val is aaaaa</b>"`;

exports[`html Named slot values fragment 1`] = `"<wc-text slot=\\"child\\">Hi</wc-text><b slot=\\"child\\">Hello</b><b slot=\\"child\\">val is foobar</b><b slot=\\"child\\">val is wow</b>"`;

exports[`html Named slot values function component 1`] = `"<b slot=\\"child\\">val is abcde</b>"`;

exports[`html Named slot values intrinsic element 1`] = `"<span slot=\\"child\\">Hello</span>"`;

exports[`html Named slot values nested class component 1`] = `"<b slot=\\"child\\">val is aaaaaaaaaaaaaaa</b>"`;

exports[`html Named slot values nested function component 1`] = `"<b slot=\\"child\\">val is abcdeabcdeabcde</b>"`;

exports[`html Named slot values null 1`] = `""`;

exports[`html Named slot values number 1`] = `"<wc-text slot=\\"child\\">123</wc-text>"`;

exports[`html Named slot values reused function component 1`] = `"<b slot=\\"child\\">val is def</b>"`;

exports[`html Named slot values text 1`] = `"<wc-text slot=\\"child\\">foobar</wc-text>"`;

exports[`html Named slot values undefined 1`] = `"<wc-text slot=\\"child\\"></wc-text>"`;

exports[`html multiple slots 1`] = `
"
      <style>
        div {
          font-size: 100px;
        }
      </style>
      <header><slot name=\\"header\\"></slot></header>
      <div><slot></slot></div>
      <header><slot name=\\"footer\\"></slot></header>
    "
`;

exports[`html multiple slots 2`] = `"<b slot=\\"header\\">head!</b><wc-text slot=\\"footer\\">foo!</wc-text><span>Foobar</span>"`;

exports[`html no slot 1`] = `
"
      <style>
        div {
          font-size: 100px;
        }
      </style>
      <div>Hello</div>
    "
`;

exports[`html no slot 2`] = `""`;

exports[`html one slot 1`] = `
"
      <style>
        div {
          font-size: 100px;
        }
      </style>
      <div><slot></slot></div>
    "
`;

exports[`html one slot 2`] = `"<span>Foobar</span>"`;
